### PR TITLE
Minor bugfixes

### PR DESCRIPTION
**Pull request by negzu**
Style submit button on password change page, fix email autocomplete in checkout, add notification when trying to add item to cart without choosing attributes

### CODE REVIEW

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- src/app/component/AddToCart/AddToCart.component.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981&bid=15703277)

File: src/app/component/AddToCart/AddToCart.style.scss [hunk 1]
```diff
@@ -63,4 +63,8 @@
             }
         }
     }
+
+    &-ButtonWrapper{
+        display:inline;
```
Comment (Danpiel) [inline src/app/component/AddToCart/AddToCart.style.scss]: Codacy Issue found: [Expected single space after ":" with a single-line declaration (declaration-colon-space-after)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.MyAccountPasswordForm{
+    &-Wrapper{
+        display: flex;
+        justify-content: center;
+        @include after-mobile{
+            padding-top: .5rem; // var(--popup-content-padding) - 1rem
+        }
+        button{
+            @include mobile{
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.MyAccountPasswordForm{
+    &-Wrapper{
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        display: flex;
+        justify-content: center;
+        @include after-mobile{
+            padding-top: .5rem; // var(--popup-content-padding) - 1rem
+        }
+        button{
+            @include mobile{
+                width:100%;
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected single space after ":" with a single-line declaration (declaration-colon-space-after)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 5]
```diff
@@ -0,0 +1,14 @@
+.MyAccountPasswordForm{
+    &-Wrapper{
+        display: flex;
+        justify-content: center;
+        @include after-mobile{
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+            padding-top: .5rem; // var(--popup-content-padding) - 1rem
+        }
+        button{
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+            @include mobile{
+                width:100%;
+            }
+        }
+    }
+}
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/AddToCart/AddToCart.style.scss [hunk 8]
```diff
@@ -63,4 +63,8 @@
             }
         }
     }
+
+    &-ButtonWrapper{
```
Comment (Danpiel) [inline src/app/component/AddToCart/AddToCart.style.scss]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss (hunk 6 above)
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 9]
```diff
@@ -0,0 +1,14 @@
+.MyAccountPasswordForm{
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/app/component/AddToCart/AddToCart.component.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981&bid=15703277)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+        @include after-mobile {
+            padding-top: .5rem; // var(--popup-content-padding) - 1rem
+        }
+        button {
+            @include mobile {
+                width: 100%;
+            }
+        }
+    }
+}
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 11]
```diff
@@ -0,0 +1,14 @@
+.MyAccountPasswordForm {
+    &-Wrapper {
+        display: flex;
+        justify-content: center;
+        @include after-mobile {
+            padding-top: .5rem; // var(--popup-content-padding) - 1rem
+        }
+        button {
```
Comment (Danpiel) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: Codacy Issue found: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4698981)

File: src/app/component/AddToCart/AddToCart.component.js [hunk 12]
```diff
@@ -52,14 +52,16 @@ export default class AddToCart extends PureComponent {
...
             isLoading,
             buttonClick,
+            disabledButtonClick,
             isDisabled
         } = this.props;
 
         if (!type_id) this.renderPlaceholder();
 
         return (
+        <div block="ProductActions-AddToCart AddToCart" elem="ButtonWrapper" onClick={isDisabled ? disabledButtonClick : () => {}}>
```
Comment (alfredsgenkins) [inline src/app/component/AddToCart/AddToCart.component.js]: Wrapper addition does not seem right to me. You should not over-complicate HTML. Please try using button only.

Comment (alfredsgenkins) [reply src/app/component/AddToCart/AddToCart.component.js]: Also, I do recommend not to add `() => {}` things like this in the code. Very misleading. It is preferable to **check for `isDisabled` in the `buttonClick` function**. Also, are you sure `isDisabled` only covers your case? I think the out of stock as well. Make sure the notification is shown only in your case.

File: src/app/component/AddToCart/AddToCart.component.js [hunk 13]
```diff
@@ -52,14 +52,16 @@ export default class AddToCart extends PureComponent {
             product: { type_id },
             isLoading,
             buttonClick,
+            disabledButtonClick,
```
Comment (alfredsgenkins) [inline src/app/component/AddToCart/AddToCart.component.js]: If new prop is added, it must be declared in prop types. I do not see this being done here.

Comment (alfredsgenkins) [reply src/app/component/AddToCart/AddToCart.component.js]: Try configuring ESLint - it will start telling you if you did something wrong :)

File: src/app/component/AddToCart/AddToCart.container.js [hunk 14]
```diff
@@ -132,6 +133,11 @@ export class AddToCartContainer extends PureComponent {
         }).then(() => this._afterAdded());
     }
 
+    disabledButtonClick(event){
+        const { showNotification } = this.props;
+        showNotification('success', 'Please, choose desired item properties');
```
Comment (alfredsgenkins) [inline src/app/component/AddToCart/AddToCart.container.js]: This is not a `success` case. More like `error`.

Comment (alfredsgenkins) [reply src/app/component/AddToCart/AddToCart.container.js]: The string you provided is not translatable. Make sure it is by wrapping in `__(<MY_STRING>)`.

Comment (alfredsgenkins) [reply src/app/component/AddToCart/AddToCart.container.js]: The string should state: `Product options are not configured!`.

Comment (alfredsgenkins) [reply src/app/component/AddToCart/AddToCart.container.js]: Please separate this PR into multiple. This is needed as it seems like some changes might not get to production - we need product owner approval.

File: src/app/component/AddToCart/AddToCart.style.scss [hunk 15]
```diff
@@ -63,4 +63,8 @@
             }
         }
     }
+
+    &-ButtonWrapper {
+        display: inline;
```
Comment (alfredsgenkins) [inline src/app/component/AddToCart/AddToCart.style.scss]: Do not forget to remove this, when removing the wrapper.

File: src/app/component/Field/Field.component.js [hunk 16]
```diff
@@ -391,6 +391,7 @@ export default class Field extends PureComponent {
               name={ name }
               value={ value }
               disabled={ isDisabled }
+              form={ form }
```
Comment (alfredsgenkins) [inline src/app/component/Field/Field.component.js]: You only added this to the `text` type. This is not correct. I still insist on input abstraction into its own component.

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.component.js [hunk 17]
```diff
@@ -39,9 +40,12 @@ class MyAccountPasswordForm extends FieldForm {
 
     renderActions() {
         return (
+        <div block="MyAccountPasswordForm" elem="Wrapper">
             <button block="Button">
                 { __('Change password') }
             </button>
+        </div>
+
```
Comment (alfredsgenkins) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.component.js]: Please do not leave ampty lines.

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.component.js [hunk 18]
```diff
@@ -39,9 +40,12 @@ class MyAccountPasswordForm extends FieldForm {
 
     renderActions() {
         return (
+        <div block="MyAccountPasswordForm" elem="Wrapper">
```
Comment (alfredsgenkins) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.component.js]: What task is this related to? Why do we need a wrapper?

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss (hunk 11 above)
Comment (alfredsgenkins) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: You can use `mix` prop on the element to mix `Button` and `MyAccountPasswordForm-Button` style. This allows to escape nesting.

File: src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss [hunk 19]
```diff
@@ -0,0 +1,14 @@
+.MyAccountPasswordForm {
+    &-Wrapper {
+        display: flex;
+        justify-content: center;
+        @include after-mobile {
+            padding-top: .5rem; // var(--popup-content-padding) - 1rem
```
Comment (alfredsgenkins) [inline src/app/component/MyAccountPasswordForm/MyAccountPasswordForm.style.scss]: I am unsure on necessity of styles above. Please check how other popups are styled. I think there is some trick to it :thinking:

File: src/app/component/ProductActions/ProductActions.style.scss [hunk 20]
```diff
@@ -137,7 +137,7 @@
 
     &-AddToCart {
         @include mobile {
-            margin: 1rem 1rem;
+            margin: 1rem .25rem;
```
Comment (alfredsgenkins) [inline src/app/component/ProductActions/ProductActions.style.scss]: Unexpected, why? :D

**Comment by alfredsgenkins** [comment]
As stated in review, please reopen this as multiple PR from feature branches.
